src/Illuminate/Routing/Controllers/HasMiddleware.php

```diff
@@ -7,7 +7,7 @@ interface HasMiddleware
     /**
      * Get the middleware that should be assigned to the controller.
      *
-     * @return \Illuminate\Routing\Controllers\Middleware[]
+     * @return array<int,\Illuminate\Routing\Controllers\Middleware|\Closure|string>
```

Comment:
Wouldn't this have sufficed?
```suggestion
     * @return (\Illuminate\Routing\Controllers\Middleware|\Closure|string)[]
```

Reply:
Also, is `string` the closest we can get? Have you tried `callable`?

Reply:
using `(...)[]` seems to work just as well but I can't say I have seen that syntax before and tend to defer to the error output from phpstan which currently uses `array<...>`

it does seem I can remove `\Closure|string` and replace with `callable` so that is a nice improvement

I have also seen `list<...>` is valid now so technically `@return list<\Illuminate\Routing\Controllers\Middleware|callable>` is an option in this case

I'll wait for github to support polls and or others to vote on the best combination and way forward :sweat_smile:

Reply:
That would actually be great for such cases 😂👍🏻